src/service/url-replacements-impl.js

```diff
@@ -586,13 +587,17 @@ export class GlobalVariableSource extends VariableSource {
    * Resolves the value via one of document info's urls.
    * @param {string} field A field on the docInfo
    * @param {string=} opt_urlProp A subproperty of the field
+   * @param {string=} opt_fallbackField A fallback field.
    * @return {T}
    * @template T
    */
-  getDocInfoUrl_(field, opt_urlProp) {
+  getDocInfoUrl_(field, opt_urlProp, opt_fallbackField) {
```

Comment:
shall we have the fallback logic in docinfo-impl?

Reply:
I'm worried that it may be confusing if a caller wants to check (say) whether there's a replace URL or not using DocInfo::replaceUrl.  I could add another field called "replaceUrlWithFallback" to avoid the confusion -- what do you think?

Reply:
I actually wanted to hide the replaceUrl concept within the docinfo. Runtime should not care if the URL is replaced or not. Unless there is a use case that we need to differentiate?

Reply:
So, in other words, the sourceUrl in docinfo would return the replace URL unless none could be found? I'm not as familiar with the runtime as you -- is it safe to do this? I.e., make it so that all requests for the source URL return the replace URL, if set?